File: Java/백준/Bronze/21312. 홀짝 칵테일/홀짝 칵테일.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] input = br.readLine().split(" ");
        int A = Integer.parseInt(input[0]);
        int B = Integer.parseInt(input[1]);
        int C = Integer.parseInt(input[2]);

        int max = A * B * C;

        //홀수만 포함된 모든 조합의 곱 중 최댓값 찾기
        int result = 0;

        if (A % 2 == 1) result = Math.max(result, A);
        if (B % 2 == 1) result = Math.max(result, B);
        if (C % 2 == 1) result = Math.max(result, C);

        if (A % 2 == 1 && B % 2 == 1) result = Math.max(result, A * B);
        if (B % 2 == 1 && C % 2 == 1) result = Math.max(result, B * C);
        if (C % 2 == 1 && A % 2 == 1) result = Math.max(result, C * A);

        if (A % 2 == 1 && B % 2 == 1 && C % 2 == 1) result = Math.max(result, A * B * C);

        if (result == 0) { //모두 짝수인 경우
            System.out.print(max);
        } else {
            System.out.print(result);
        }
    }
}

File: C11/백준/Bronze/21312. 홀짝 칵테일/홀짝 칵테일.c
#include <stdio.h>

int max_val(int a, int b) {
	return a > b ? a : b;
}

int main(void) {
	
	int A, B, C;
	scanf("%d %d %d", &A, &B, &C);
	
	int max = A * B * C;
	int result = 0;
	
	if (A % 2 == 1) result = max_val(result, A);
	if (B % 2 == 1) result = max_val(result, B);
	if (C % 2 == 1) result = max_val(result, C);
	
	if (A % 2 == 1 && B % 2 == 1) result = max_val(result, A * B);
	if (B % 2 == 1 && C % 2 == 1) result = max_val(result, B * C);
	if (C % 2 == 1 && A % 2 == 1) result = max_val(result, C * A);
	
	if (A % 2 == 1 && B % 2 == 1 && C % 2 == 1) result = max_val(result, A * B * C);
	
	if (result == 0) {
		printf("%d", max);
	} else {
		printf("%d", result);
	}
	
	return 0;
}